CONTRIBUTING.md

```diff
@@ -4,22 +4,38 @@
 
 If you would like to contribute enhancements or fixes, please do the following:
 
-1.  Fork the package repository.
-2.  Run `npm install` to setup all dependencies.
-3.  Hack on a separate topic branch created from the latest `master`.
-4.  Check for lint errors with `npm run lint` or use `linter-eslint` within Atom.
-5.  Run the package specs.
-6.  Commit the changes and push the topic branch to your fork.
-7.  Make a pull request to the main repo.
-8.  Welcome to the club!
+1. Fork the package repository and clone your fork to local file.
+
+1. Run `npm install` in the project to setup all dependencies.
+
+1. Run `apm link --dev` to install your working folder as a development-mode-only package in Atom.
+
+1. While on the `master` branch run `git checkout -b my-super-special-branch` to create and checkout a separate branch for your work. This makes merging your changes a lot easier for the maintainers.
```

Comment:
Maybe add a note about bringing in changes to `master` while you are working on `my-super-special-branch`?

Reply:
I'm not sure what you're discussing here. Is this about rebasing on master branch to stay up to date?

Reply:
Yea, I'll see people submit PR's directly from their `master` and then get into _huge_ messes when their PR doesn't get merged before `master` gets changed.

Reply:
So what instruction would you suggest on this? I mean you can rebase directly off `upstream master` to a branch the same as rebasing off a local `master`. If there is a merge conflict, it's gonna bite you either way. And merging instead of rebasing adds useless merge commits.

Reply:
Yea, for somebody familiar with `git` it's not really that different. For everyone that I've seen submitting a PR off of `master`? It's incredibly difficult. Then again they are unlikely to read the instructions here so...

Anyway, I guess I'd write something like this?
```markdown
While on the `master` branch run `git checkout -b my-super-special-branch` to create and checkout a separate branch for your work. This makes it easier to rebase your work later if `master` updates before it gets merged.
```